controllers/authController.js

```diff
@@ -64,3 +79,73 @@ exports.loginUser = (req, res) => {
         return res.status(200).json({message: "Successfully logged in", token})
   })
 }
+
+//This function is responsible for sending email to a particular user with the token for password reset
+exports.forgotPassword = (req, res) => {
+  User.findOne({email: req.body.email} , (err, foundUser) => {
+    if (err) {
+      return res.status(400).json({err})
+    }
+    if (!foundUser) {
+      return res.status(401).json({message: "User with this email does not exist"})
+    }
+    let token = await Token.findOne({ userId: user._id});
+    if (!token) {return res.status(400).send("Invalid user");}
+    token = await new Token({
```

Comment:
instead of using `await new Token ({...})` and then `.save()`, you can use `await Token.create({...})`